src/modules/ibc/impls.rs

```diff
@@ -79,8 +77,18 @@ use tendermint_proto::{
 };
 use tracing::debug;
 
+use derive_more::{From, TryInto};
+
 use ibc::core::dispatch;
 
+// Note: We define `AnyConsensusState` just to showcase the use of the
+// derive macro. Technically, we could just use `TmConsensusState`
+// as the `AnyConsensusState`, since we only support this one variant.
+#[derive(ConsensusState, From, TryInto)]
+pub enum AnyConsensusState {
+    Tendermint(TmConsensusState),
+}
```

Comment:
I've tried out the impls with `TmConsensusState` to learn how things work around. The existing [`TryInto`](https://github.com/cosmos/ibc-rs/blob/7e921a4a9f2e334e6cc5cfe54b05a07f1abf3bb4/crates/ibc/src/clients/ics07_tendermint/context.rs#L15) bound under the CommonContext preventing to do that. Am I missing something?

Reply:
[0cb3657](https://github.com/cosmos/ibc-rs/pull/683/commits/0cb3657ebb1c9787348d465add2f0a90d6256013) fixes this. If you want to use `TmConsensusState` directly, you need to set `CommonContext::ConversionError = Infaillible` and it will work. Good catch!